app/services/dqt/sync_teacher.rb

```diff
@@ -30,18 +30,19 @@ def call
                                     eyts_date: dqt_data["eytsDate"],
                                     early_years_status_name: dqt_data.dig("earlyYearsStatus", "name"),
                                     early_years_status_value: dqt_data.dig("earlyYearsStatus", "value"))
-      dqt_teacher.save
 
-      dqt_data["initialTeacherTraining"].each do |training_data|
-        dqt_teacher_training = TeacherTraining.find_or_initialize_by(dqt_teacher:)
-        dqt_teacher_training.assign_attributes(programme_start_date: training_data["programmeStartDate"],
-                                               programme_end_date: training_data["programmeEndDate"],
-                                               programme_type: training_data["programmeType"],
-                                               result: training_data["result"],
-                                               hesa_id: training_data["husId"], # associated with that period of training
-                                               provider_ukprn: training_data.dig("provider", "ukprn"))
-        dqt_teacher_training.save
+      dqt_teacher.dqt_trainings = dqt_data["initialTeacherTraining"].map do |training_data|
```

Comment:
How does this behave when the `Teacher` already exists with `dqt_trainings`? Will the old records be deleted, overwritten or be orphaned?